crates/rolldown/src/ecmascript/format/iife.rs

```diff
@@ -37,25 +38,30 @@ pub fn render_iife(
   let export_items = get_export_items(ctx.chunk, ctx.link_output);
   let has_exports = !export_items.is_empty();
   let has_default_export = export_items.iter().any(|(name, _)| name.as_str() == "default");
-  // Since before rendering the `determine_export_mode` runs, `unwrap` here won't cause panic.
-  // FIXME do not call `determine_export_mode` twice
   let entry_module = match ctx.chunk.kind {
     ChunkKind::EntryPoint { module, .. } => {
       &ctx.link_output.module_table.modules[module].as_ecma().expect("should be ecma module")
     }
     ChunkKind::Common => unreachable!("iife should be entry point chunk"),
   };
-  let export_mode = determine_export_mode(&ctx.options.exports, entry_module, &export_items)?;
+  let export_mode = determine_export_mode(ctx, entry_module, &export_items)?;
   let named_exports = matches!(&export_mode, OutputExports::Named);
 
   let (import_code, externals) = render_iife_chunk_imports(ctx);
 
   let (input_args, output_args) =
-    render_iife_arguments(&externals, &ctx.options.globals, has_exports && named_exports);
+    render_iife_arguments(ctx, &externals, &ctx.options.globals, has_exports && named_exports);
```

Comment:
Since you already pass the `ctx`, you don't need to pass `ctx.options.globals`.

Reply:
Repaired.